feat(minifier): disable removal of unnecessary `use strict` directives for DCE

sapphi-red opened the pull request:
This is kind of a workaround. The target of this PR is to solve https://github.com/rolldown/rolldown/issues/6879 and https://github.com/rolldown/rolldown/issues/6880.

While the proper way to fix them is to use the proper `ModuleKind` (described in https://github.com/rolldown/rolldown/issues/7009), `ModuleKind::Unambiguous` doesn't work as expected and is blocked by https://github.com/oxc-project/oxc/issues/15541.

So for the meanwhile, this PR can solve those issues indirectly. We can revert this PR once the issues above are solved properly, but I guess it's also fine to keep it as I think there's less value to remove `use strict` directives in DCE.

sapphi-red commented:
* **#15691** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15691?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15691?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=15691).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Nov 15, 3:40 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 15, 3:40 AM UTC**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

sapphi-red committed:
feat(minifier): disable removal of unnecessary `use strict` directives for DCE (#15691)

This is kind of a workaround. The target of this PR is to solve https://github.com/rolldown/rolldown/issues/6879 and https://github.com/rolldown/rolldown/issues/6880.

While the proper way to fix them is to use the proper `ModuleKind` (described in https://github.com/rolldown/rolldown/issues/7009), `ModuleKind::Unambiguous` doesn't work as expected and is blocked by https://github.com/oxc-project/oxc/issues/15541.

So for the meanwhile, this PR can solve those issues indirectly. We can revert this PR once the issues above are solved properly, but I guess it's also fine to keep it as I think there's less value to remove `use strict` directives in DCE.
